tools/auth/spotify.go

```diff
@@ -0,0 +1,50 @@
+package auth
+
+import (
+	"log"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/spotify"
+)
+
+var _ Provider = (*Spotify)(nil)
+
+// NameSpotify is the unique name of the Spotify provider.
+const NameSpotify string = "spotify"
+
+// Spotify allows authentication via Spotify OAuth2.
+type Spotify struct {
+	*baseProvider
+}
+
+// NewSpotifyProvider creates a new Spotify provider instance with some defaults.
+func NewSpotifyProvider() *Spotify {
+	return &Spotify{&baseProvider{
+		scopes:     []string{"user-read-private", "user-read-email"},
+		authUrl:    spotify.Endpoint.AuthURL,
+		tokenUrl:   spotify.Endpoint.TokenURL,
+		userApiUrl: "https://api.spotify.com/v1/me",
+	}}
+}
+
+// FetchAuthUser returns an AuthUser instance based on the Spotify's user api.
+func (p *Spotify) FetchAuthUser(token *oauth2.Token) (*AuthUser, error) {
+	rawData := struct {
+		Id    string `json:"id"`
+		Name  string `json:"display_name"`
+		Email string `json:"email"`
+	}{}
```

Comment:
I'm working on these changes now!

I was wondering if you would like me add the changes in `ui/dist` after running
```npm run build```
inside the `ui` directory?

Right now I am painstakingly restoring every added/deleted/changed file in `ui/dist` to make the patch easier to review - but maybe I should just commit everything?

Reply:
You can commit them safely with the other changes and I'll just ignore the generated files.

_The goreleaser action generates its own `ui/dist` build and it will be compared with the one in the branch and in case of a difference (eg. missing/different js dependency or something else) the action will fail._